test/e2e/tests/add-custom-network.spec.js

```diff
@@ -398,21 +405,13 @@ describe('Custom network', function () {
           tag: 'div',
         });
 
-        await driver.clickElement({
-          tag: 'button',
-          text: 'Delete',
-        });
+        // Click first Delete button
+        await driver.clickElement('button.btn-danger');
```

Comment:
Nit: This seems like a step in the wrong direction. The test is now more brittle because it depends upon the specific tag and classname used, which might change. It's also less readable (it's not clear this is a "Delete" button, but it was clear beforehand).

Generally I recommend using just a `text` selector if possible, with a `data-testid` selector as well if necessary. Text selectors are better for readability in that it makes the test more closely match what a user sees when going through this flow, and they're better for maintainability because the text won't change during a refactor. `data-testid` isn't as readable, but it's also nice for maintainability because it flags in the code where a test relies upon an element.

I'm curious what motivated the change - was there a second "Delete" button on the page somewhere? Or was this clicking "Delete" before the Aritrum page rendered (i.e. when the previous network details were shown)?

Reply:
Yes, there are two "Delete" buttons in a row, and I was worried about the driver getting them confused.

![image](https://github.com/MetaMask/metamask-extension/assets/539738/23807799-e5a7-42b5-a2c0-6f8ce51c9131) ![image](https://github.com/MetaMask/metamask-extension/assets/539738/8acffc80-e949-4624-879d-f7348f9ef8e6)

Reply:
Ahh I see. Hmm. I guess a data-testid could work to differentiate between them